cli/silence_update.go

```diff
@@ -98,30 +100,43 @@ func getSilenceById(silenceId string, baseUrl url.URL) (*types.Silence, error) {
 }
 
 func updateSilence(silence *types.Silence) (*types.Silence, error) {
-	if *updateExpires != "" {
-		duration, err := model.ParseDuration(*updateExpires)
+	var err error
+	if *updateEnd != "" {
+		silence.EndsAt, err = time.Parse(time.RFC3339, *updateEnd)
 		if err != nil {
 			return nil, err
 		}
-		if duration == 0 {
+	} else if *updateDuration != "" {
+		d, err := model.ParseDuration(*updateDuration)
+		if err != nil {
+			return nil, err
+		}
+		if d == 0 {
 			return nil, fmt.Errorf("silence duration must be greater than 0")
 		}
-		silence.EndsAt = time.Now().UTC().Add(time.Duration(duration))
+		silence.EndsAt = silence.EndsAt.UTC().Add(time.Duration(d))
```

Comment:
It is now adding the passed duration to the existing `EndsAt`. I suppose this is intentional but it wasn't obvious from the change description.

Reply:
Hm, yeah, the original behavior was confusing to me. Now that I'm looking at this again, I'm thinking the duration should be added to `StartsAt`...

With `amtool update <id> --start=<time> --duration=2h`, I would expect this silence to be active for 2 hours after the start time.
With `amtool update <id> --duration=2h` I think it's less clear what the behavior should be. I would take this to mean "set the duration to `start_time+2h`, but I'm not sure if other folks agree?

Reply:
Yeah my gut feeling is that `amtool update <id> --duration=<x>` is ambiguous for users whichever time reference is chosen. Using `StartsAt` is probably the less confusing option but this is just my opinion ;-)